### ISSUE TITLE
Improve accessibility of resize actions in Datagrid

### ISSUE THREAD

**Issue by elycheea** [carbon-design-system/ibm-products#3518]
The original accessibility guidance for resizable columns was shared as follows:

> #### Keyboard navigation
> - [ ] <kbd>Tab</kbd>: navigates the user to the different interactive elements
> - [ ] <kbd>Space</kbd> or <kbd>Enter</kbd>: triggers the selected element
> - [ ] <kbd>Left</kbd> & <kbd>Right</kbd> arrow keys: resize column left or right
>
> - [ ] Tab order
      - First interactive element in the body area
      - Proceed left to right and top to bottom through the rest of the body elements.
      - Primary cell
      - Secondary resize handle
      - Secondary cell

Mike Gower brought up a number of concerns on resizing columns based on pointer accessibility and screen reader accessibility.

We should work with IBM Accessibility on this based on a number of comments from Mike Gower in #1923.

In summary, the suggestions from Mike include
1. [Using a menu](https://github.com/carbon-design-system/ibm-products/issues/1923#issuecomment-1302253470)
    > Imagine if [row height settings] name was "Table properties" and one section or submenu inside it was "row height" (to keep providing your existing function). We could have another section or submenu for "column width" and it could contain things like "minimum", "reset", "maximum","resize to fit data", and "set to...". It could also potentially have its own submenu for each of the columns.
2. [Incorporating keyboard interaction into the column headers](https://github.com/carbon-design-system/ibm-products/issues/1923#issuecomment-1302314847)
   Expect this approach to be closest to our originally intended design, but does have its own challenges and potential conflicts with other extensions / Datagrid functionality.
3. [Using header cell actions](https://github.com/carbon-design-system/ibm-products/issues/1923#issuecomment-1302335488) 
    > I thought I'd mention this potential approach. The user **activates a header cell**, at which point all the potential functions of the cell are revealed. There are a few ways to provide this. One would be making each cell effectively a menu button.
You could potentially treat your header row as a single complex component. The whole row might only need to take one tab stop, with the user navigating into it to activate different cells. I'm not going to go into this in detail, since it's obviously quite a departure. But wanted to flag that there might be other ways to approach.

Of course, a number of these do begin teetering into the business logic / product implementation side but we should be able to support these as intended behaviors if we suggest them as alternatives.

**Comment by elycheea** [comment]
@vsnichols This might be something we need to re-evaluate if you can rally some designers to help review these considerations.
